cli/src/config.rs

```diff
@@ -86,6 +100,30 @@ where
     Ok(Option::<bool>::deserialize(deserializer)?.unwrap_or_default())
 }
 
+/// Load the configuration file.
+pub fn load_config(matches: &ArgMatches) -> Result<Config> {
+    // Early return with configuration disabled.
+    if matches.get_flag("no-config") {
+        return Ok(Config::default());
+    }
+
+    let settings_path = get_home_settings_path()?;
+    let config_path = matches
+        .get_one::<String>("config")
+        .and_then(|config_path| shellexpand::env(config_path).ok())
+        .map(|config_path| PathBuf::from(config_path.to_string()))
+        .unwrap_or(settings_path);
+
+    log::debug!("Reading config from {}", config_path.to_string_lossy());
+    let mut config: Config = read_configuration(&config_path).map_err(|err| {
+        anyhow!("Failed to read configuration at `{}`: {}", config_path.to_string_lossy(), err)
+    })?;
```

Comment:
I know this code was only moved. But I can't help but notice the `.map_err()` here should probably be `.with_context()` instead.

```suggestion
    let mut config: Config = read_configuration(&config_path).with_context(|| {
        format!("Failed to read configuration at `{}`", config_path.to_string_lossy())
    })?;
```